src/hotspot/share/classfile/verifier.cpp

```diff
@@ -3149,7 +3149,7 @@ void ClassVerifier::verify_return_value(
   if (return_type == VerificationType::bogus_type()) {
     verify_error(ErrorContext::bad_type(bci,
         current_frame->stack_top_ctx(), TypeOrigin::signature(return_type)),
-        "Method expects a return value");
+        "Method does not expect a return value");
```

Comment:
You've changed the wrong line. This one does expect a return value

Reply:
Hi David,
If a method's descriptor has a void return type, but the method contains a bytecode such as areturn, then I think the message "Method does not expect a return type" is correct.  Perhaps the word 'expect' is ambiguous?  If so, then I think a more explicit message, similar to my original suggestion, should be used.
Harold

Reply:
Okay I see now the original bug report was incorrect and the referenced code is correct. But the above code is actually incorrect and now fixed.